### PR TITLE
fix: improve support for Blob and File like objects

### PR DESCRIPTION
**Pull request by ronag**
Refs: https://github.com/nodejs/undici/pull/1109

### CODE REVIEW

**Review by mcollina — COMMENTED** [review]
This needs tests

**Comment by ronag** [comment]
@wojpawlik

**Comment by wojpawlik** [comment]
```
~/JS/client (remove-hacks *$%=)
$ npm i --no-save https://github.com/nodejs/undici/tree/blob-file-like

changed 1 package, and audited 4 packages in 1m     

found 0 vulnerabilities

~/JS/client (remove-hacks *$%=)
$ node tst.cjs
ReferenceError: isFileLike is not defined
    at makeEntry (D:\Wojciech\JS\client\node_modules\undici\lib\fetch\formdata.js:235:3)
```

**Review by mcollina — APPROVED** [review]
lgtm

File: lib/fetch/util.js [hunk 1]
```diff
@@ -74,13 +75,20 @@ function isBlobLike (object) {
...
     /^(Blob|File)$/.test(object[Symbol.toStringTag])
   )
 }
 
+function isFileLike (object) {
+  return object instanceof File || (
+    object &&
+    typeof object === 'object' &&
+    (typeof object.stream === 'function') &&
+    /^(File)$/.test(object[Symbol.toStringTag])
```
Comment (wojpawlik) [inline lib/fetch/util.js:88]:
```suggestion
    object[Symbol.toStringTag] === 'File'
```

**Review by wojpawlik — APPROVED** [review]
Thanks, this fixes my issue!

**Pull request merged**